src/Bitcoin/Transaction/Common.hs

```diff
@@ -143,77 +138,67 @@ data Tx = Tx
 
 -- | Compute transaction hash.
 txHash :: Tx -> TxHash
-txHash tx = TxHash . doubleSHA256 . runPutS $ serialize tx{txWitness = []}
+txHash tx = TxHash . doubleSHA256L $ Bin.encode tx{txWitness = []}
 
 
 instance IsString Tx where
     fromString =
-        fromMaybe e . (eitherToMaybe . runGetS deserialize <=< decodeHex) . cs
+        fromMaybe e . (eitherToMaybe . U.decode . BSL.fromStrict <=< decodeHex) . cs
       where
         e = error "Could not read transaction from hex string"
 
 
-instance Serial Tx where
-    deserialize =
+instance Binary Tx where
+    get =
         isWitnessTx >>= \w -> if w then parseWitnessTx else parseLegacyTx
-    serialize tx
+    put tx
         | null (txWitness tx) = putLegacyTx tx
         | otherwise = putWitnessTx tx
 
 
-instance Binary Tx where
-    put = serialize
-    get = deserialize
-
-
-instance Serialize Tx where
-    put = serialize
-    get = deserialize
-
-
-putInOut :: MonadPut m => Tx -> m ()
+putInOut :: Tx -> Put
 putInOut tx = do
     putVarInt $ length (txIn tx)
-    forM_ (txIn tx) serialize
+    forM_ (txIn tx) put
     putVarInt $ length (txOut tx)
-    forM_ (txOut tx) serialize
+    forM_ (txOut tx) put
 
 
--- | Non-SegWit transaction serializer.
-putLegacyTx :: MonadPut m => Tx -> m ()
+-- | Non-SegWit transaction putr.
```

Comment:
Not sure if the comment mutations `serializer -> putr` and `deserializer -> getr` are intentional.

Reply:
Looks like some casualties of my search/replace procedure!